src/topology/metric_space/cau_seq_filter.lean

```diff
@@ -6,327 +6,16 @@ Characterize completeness of metric spaces in terms of Cauchy sequences.
-  -- It is Cauchy.
-  have : cauchy_seq u := sequentially_complete.seq_of_cau_filter_is_cauchy hf (λn, 1) (λn, ennreal.zero_lt_one),
-  -- Therefore, it converges by assumption. Let `x` be its limit.
-  rcases H u this with ⟨x, hx⟩,
-  -- The original filter also converges to `x`.
-  exact ⟨x, sequentially_complete.le_nhds_cau_filter_lim hf (λn, 1) (λn, ennreal.zero_lt_one) hx⟩
-end⟩
-
-/-- A very useful criterion to show that a space is complete is to show that all sequences
-which satisfy a bound of the form `edist (u n) (u m) < B N` for all `n m ≥ N` are
-converging. This is often applied for `B N = 2^{-N}`, i.e., with a very fast convergence to
-`0`, which makes it possible to use arguments of converging series, while this is impossible
-to do in general for arbitrary Cauchy sequences. -/
-theorem emetric.complete_of_convergent_controlled_sequences {α : Type u} [emetric_space α]
-  (B : ℕ → ennreal) (hB : ∀n, 0 < B n)
-  (H : ∀u : ℕ → α, (∀N n m : ℕ, N ≤ n → N ≤ m → edist (u n) (u m) < B N) → ∃x, tendsto u at_top (𝓝 x)) :
-  complete_space α :=
-⟨begin
-  -- Consider a Cauchy filter `f`.
-  intros f hf,
-  -- Introduce a sequence `u` approximating the filter `f`.
-  let u := sequentially_complete.seq_of_cau_filter hf B hB,
-  -- It satisfies the required bound.
-  have : ∀N n m : ℕ, N ≤ n → N ≤ m → edist (u n) (u m) < B N := λN n m hn hm, calc
-    edist (u n) (u m) < sequentially_complete.B2 B N :
-      sequentially_complete.seq_of_cau_filter_bound hf B hB hn hm
-    ... ≤ B N : lattice.inf_le_right,
-  -- Therefore, it converges by assumption. Let `x` be its limit.
-  rcases H u this with ⟨x, hx⟩,
-  -- The original filter also converges to `x`.
-  exact ⟨x, sequentially_complete.le_nhds_cau_filter_lim hf B hB hx⟩
-end⟩
-
-/-- A very useful criterion to show that a space is complete is to show that all sequences
-which satisfy a bound of the form `dist (u n) (u m) < B N` for all `n m ≥ N` are
-converging. This is often applied for `B N = 2^{-N}`, i.e., with a very fast convergence to
-`0`, which makes it possible to use arguments of converging series, while this is impossible
-to do in general for arbitrary Cauchy sequences. -/
-theorem metric.complete_of_convergent_controlled_sequences {α : Type u} [metric_space α]
-  (B : ℕ → real) (hB : ∀n, 0 < B n)
-  (H : ∀u : ℕ → α, (∀N n m : ℕ, N ≤ n → N ≤ m → dist (u n) (u m) < B N) → ∃x, tendsto u at_top (𝓝 x)) :
-  complete_space α :=
-begin
-  -- this follows from the same criterion in emetric spaces. We just need to translate
-  -- the convergence assumption from `dist` to `edist`
-  apply emetric.complete_of_convergent_controlled_sequences (λn, ennreal.of_real (B n)),
-  { simp [hB] },
-  { assume u Hu,
-    apply H,
-    assume N n m hn hm,
-    have Z := Hu N n m hn hm,
-    rw [edist_dist, ennreal.of_real_lt_of_real_iff] at Z,
-    exact Z,
-    exact hB N }
-end
-
-section
-
 /- Now, we will apply these results to `cau_seq`, i.e., "Cauchy sequences" defined by a
```

Comment:
This doctring feels weird now that the file is empty before it...

Reply:
Fixed.